code_annotations/cli.py

```diff
@@ -1,20 +1,71 @@
 """
 Command line interface for code annotation tools.
 """
+import os
+import sys
+
 import click
+import yaml
+from code_annotations.django_reporting_helpers import get_models_requiring_annotations
+
+DEFAULT_SAFELIST_FILE_PATH = '.pii_safe_list.yaml'
+
+
+def fail(msg):
+    """
+    Log the message and exit.
+    """
+    click.echo(msg)
+    sys.exit(1)
 
 
 @click.group()
-def cli():
+@click.option(
+    '--config-file', 'config_file_path',
+    type=click.Path(exists=True, dir_okay=False),
+    help='Config file for all code annotation tools.',
+)
+@click.pass_context
+def cli(ctx, config_file_path):
     """
     Code annotation tools.
     """
-    pass
+    # Code placed in this function runs first, before entering any subcommands.
+    ctx.ensure_object(dict)
+    with open(config_file_path) as config_file:
+        ctx.obj['config'] = yaml.load(config_file)
+    # Now, the ctx object will be passed along to subcommands as the first argument.
 
 
-@cli.command('seed_safelist')
-def seed_safelist():
+@cli.command('pii_report_django')
+@click.option(
+    '--seed-safelist',
+    is_flag=True,
+    help='Generate an initial safelist file based on the current Django environment.',
+)
+@click.pass_context
+def pii_report_django(ctx, seed_safelist):
     """
-    Subcommand for seeding the initial .pii_safe_list.yaml file.
+    Subcommand for dealing with PII in Django models.
```

Comment:
Here's an example console log from running this command:

```
Found 2 non-local models requiring annotations. Adding them to safelist.
Found 2 local models requiring annotations. NOT adding them to safelist.
Successfully created safelist file "fake_safelist_path.yaml".
Now, you need to:
  1) Make sure that any un-annotated models in the safelist are annotated, and
  2) Annotate the following local models:
     fake_app_1.FakeModelA
     fake_app_2.FakeModelB
```

Reply:
For the current iteration, I simply pushed the list up before the important log messages.

I didn't want to pollute the safelist file, and writing out a second file became complicated considering that it would probably need to go into a `log/` directory so that git will not add it to the index, and it would probably need some sort of log rotation logic to avoid clobbering itself.

Reply:
```
Listing 2 local models requiring annotations:
     fake_app_1.FakeModelA
     fake_app_2.FakeModelB
Found 2 non-local models requiring annotations. Adding them to safelist.
Found 2 local models requiring annotations. NOT adding them to safelist.
Successfully created safelist file "fake_safelist_path.yaml".
Now, you need to:
  1) Make sure that any un-annotated models in the safelist are annotated, and
  2) Annotate the local models listed above.
```

Reply:
I think this is ok for now based on the Slack discussion, but we should make a ticket to move it out to a separate command.

Reply:
PLAT-2392